### DISCUSSION TITLE
Supporting onSomething={[fn, arg]} on my components

### DISCUSSION THREAD

**Discussion by tslocke** [solidjs/solid#975] [Q&A]
For native events (from docs):

> Solid also supports passing an array to the event handler to bind a value to the first argument of the event handler. This doesn't use bind or create an additional closure, so it is a highly optimized way of delegating events.

I'd like to support this for my own events, e.g. `<PopupMenuItem action={[fn, arg]}/>`

I figured there might be something in the Solid API like `callHandler(x, event)` that would check if it's a function or an array, and invoke it either way, but I can't find anything. Does it exist?

Not a big deal to write myself but I'd prefer to do it the 'official' way if there is one.

**Comment by gerardkd** [comment]
If I'm reading your question right, this is what you're looking for: [Playground link](https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EogcuPfrwAKtQvUIBZYfQCSuOIulyCspgAHT6XC6EWvAQ9FpovjAWVlYy9BDMuGi0ELxu6BAAFACU-Fa8vMw5ZHw5AIKZ2bkAvLyFvhS4iLw1jBgA5li8cABuwl28AKJjELilzQB8ZbkVldXqcCFqtP2FKlCNOZUazADWcCIqQwIduEOj41LFliuSL+WCcLhMuQA8TpEPLEfH5eAcsjlmiASrxFrx6ocINJ9AsXpIUhAhKJxIUYXDfvkMLwUUMRLRmPQYrgQv0vpM1H5xgAhfBeER7MBQQiEFTFACEzysYEkWHA0HgSDAEVc0U88US1AIxEl5CoODoTFYkps3D4AgAwgouBBHgF5IoVHQNNo9MkIFY4BRbHwqhAao5nK4gd5Et1DZwcuNfgJwU1ugAxdKIqRLVqFQjyQhkeZLEAfITfRi5PYrCq-ERoEYLD6rXi-QgLAFezwgmC-cLF3OrX4mei4XBHHL6k6nKHABPOMghUM5EpDADkAAs4GptrxDNw1CJxwBdSQLLwQEa0c5gxH11vtnKN0v1gtFj6CpEvYUroA).

You can add a prop to the `PopupMenuItem` component that is a function and call that function from within your child component. In your parent component, you can then assign a function to the `action` prop on `<PopMenuItem action={() => onAction} />`.

**Comment by otonashixav** [comment]
There is no function which handles "bound" event handlers. The relevant code is here: https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/client.js#L83-L92 and https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/client.js#L305-L340

So you would have to write it yourself.
